plugins/scripting/scripting-compiler/src/org/jetbrains/kotlin/scripting/compiler/plugin/impl/errorReporting.kt

```diff
@@ -50,6 +51,37 @@ class ScriptDiagnosticsMessageCollector(private val parentMessageCollector: Mess
     }
 }
 
+class ScriptDiagnosticsMessageExtendedCollector(parentMessageCollector: MessageCollector?) :
```

Comment:
`Extended` part of the name sounds weird to me. It seems that there is no good reason to have a separate "extended" variant, and you can simply add this functionality to the `ScriptDiagnosticsMessageCollector`.